pkg/controller/acmeorders/sync.go

```diff
@@ -514,7 +524,7 @@ func (c *controller) finalizeOrder(ctx context.Context, cl acmecl.Interface, o *
 	// if it is already in the 'valid' state, as upon retry we will
 	// then retrieve the Certificate resource.
 	_, errUpdate := c.updateOrderStatus(ctx, cl, o)
-	if acmeErr, ok := err.(*acmeapi.Error); ok {
+	if acmeErr, ok := errUpdate.(*acmeapi.Error); ok {
```

Comment:
You've not updated the `o.Status.Reason = fmt.Sprintf("Failed to retrieve Order resource: %v", err)` line below here when you changed this, so here we are checking `errUpdate` but will print the contents of `err` instead.